src/defs.h

```diff
@@ -131,25 +160,30 @@ typedef struct {
     int disabled;
 } macro_t;
 
+typedef struct fn fn_t;
```

Comment:
It is confusing to have two derived types, `fn_t` and `func_t`. Can you clarify?

Reply:
`typedef` is useful if we want to emphasize the types. However, it may become redundant once it is referred to be a shortcut of structures.

Reply:
> It is confusing to have two derived types, fn_t and func_t. Can you clarify?

Those two one should be the same structure, but the difference here is that one is declared to a static array and the other is dynamically allocated. I keep the old one for the compatibility to @ChAoSUnItY 's works.

Reply:
> Those two ones would be the same structure, but the difference here is one is declared to a static array and the other is dynamically allocated. I keep the old one for the compatibility to @ChAoSUnItY 's works.

Add comments with corresponding FIXME/TODO notation.